### ISSUE TITLE
XML project file format

### ISSUE THREAD

**Issue by o-jill** [o-jill/kouets#19]
how about change project file format to XML?
it is useful for adding some parameters.
how to use SAX parser in Qt:
- http://doc.qt.io/qt-4.8/xml-sax.html
- http://www.informit.com/articles/article.aspx?p=1405553&seqNum=3

**Comment by o-jill** [comment]
you can also choose JSON as  a file format :smiley:

**Comment by o-jill** [comment]
this is an example.

``` xml
<?XML version="1.0" ?>
<kouets version="1.0">
 <config>
  <activatetab> FALSE </activatetab>
  <apppath> C:/path/to/default.exe </apppath>
  <cmdline> --some --default --command --option </cmdline>
  <color1> #FF0000 </color1>
 </config>
 <item>
  <file>kouetsapp.cpp</file>
  <!-- default program(apppath) and cmdline  will be used -->
 </item>
 <item>
  <file> kouetsapp.h </file>
  <apppath> C:/speci/alpat/htoan/application.exe </program>
  <cmdline> --some --command --option </cmdline>
  <parser> cpplint.py_VS7 </parser>
 </item>
</kouets>
```

**Comment by o-jill** [comment]
this is JSON example
but unfortunately JSON sometimes has comma problem...

``` json
{
 "kouets": {
  "config": {
   "activetab": "FALSE",
   "apppath": "C:/path/to/default.exe",
   "cmdline": "--some --default --command --option"
  },
  "item": {
   "file":"kouetsapp.cpp"
  },
  "item": {
   "file":"kouetsapp.h",
   "apppath":"C:/speci/alpat/htoan/application.exe",
   "cmdline":"--some --command --option",
   "parser":"cpplint.py_VS7"
  }
 }
}
```

**Comment by o-jill** [comment]
file format specification was written in projectfilespec.md.

**Comment by o-jill** [comment]
implemented in PR #29.
